server/src/main/java/org/elasticsearch/index/codec/PerFieldFormatSupplier.java

```diff
@@ -58,9 +59,9 @@ public PostingsFormat getPostingsFormatForField(String field) {
 
     private PostingsFormat internalGetPostingsFormatForField(String field) {
         if (mapperService != null) {
-            final PostingsFormat format = mapperService.mappingLookup().getPostingsFormat(field);
-            if (format != null) {
-                return format;
+            Mapper mapper = mapperService.mappingLookup().getMapper(field);
+            if (mapper instanceof CompletionFieldMapper) {
+                return CompletionFieldMapper.postingsFormat();
```

Comment:
@rjernst asked me to clean this up to work similarly to `getKnnVectorsFormatForField` below. Now, the `CompletionFieldMapper` is solely responsible for exposing it's own postings format.

Reply:
+1 to following up with moving the contents of `CompletionFieldMapper.postingsFormat()` into here.

Reply:
I suppose I have some confusion as the the purview of each class with respect to who ultimately is the source of truth for the format.  To me it seems like the mapper should own it, though I see mixed examples of ownership here.  I however agree it's cleaner to just move it here, so I'll do that.